applications/tari_console_wallet/src/init/mod.rs

```diff
@@ -0,0 +1,344 @@
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::{
+    utils::db::get_custom_base_node_peer_from_db,
+    wallet_modes::{PeerConfig, WalletMode},
+};
+use log::*;
+use rpassword::prompt_password_stdout;
+use std::{fs, str::FromStr, sync::Arc};
+use tari_app_utilities::utilities::{setup_wallet_transport_type, ExitCodes};
+use tari_common::{ConfigBootstrap, GlobalConfig, Network};
+use tari_comms::{peer_manager::Peer, NodeIdentity};
+use tari_comms_dht::{DbConnectionUrl, DhtConfig};
+use tari_core::{consensus::Network as NetworkType, transactions::types::CryptoFactories};
+use tari_p2p::{initialization::CommsConfig, seed_peer::SeedPeer, DEFAULT_DNS_SEED_RESOLVER};
+use tari_shutdown::ShutdownSignal;
+use tari_wallet::{
+    base_node_service::config::BaseNodeServiceConfig,
+    error::{WalletError, WalletStorageError},
+    output_manager_service::config::OutputManagerServiceConfig,
+    storage::sqlite_utilities::initialize_sqlite_database_backends,
+    transaction_service::config::TransactionServiceConfig,
+    wallet::WalletConfig,
+    Wallet,
+    WalletSqlite,
+};
+
+pub const LOG_TARGET: &str = "wallet::console_wallet::init";
+/// The minimum buffer size for a tari application pubsub_connector channel
+const BASE_NODE_BUFFER_MIN_SIZE: usize = 30;
+const TARI_WALLET_PASSWORD: &str = "TARI_WALLET_PASSWORD";
+
+pub fn get_or_prompt_password(
+    arg_password: Option<String>,
+    config_password: Option<String>,
+) -> Result<Option<String>, ExitCodes>
+{
+    if arg_password.is_some() {
+        return Ok(arg_password);
+    }
+
+    let env = std::env::var_os(TARI_WALLET_PASSWORD);
```

Comment:
This is not needed, I think the env var `TARI_WALLET__RIDCULLY__PASSWORD` should already handle it

Reply:
Oh? I’ll take a look for that